tests/dragonfly/tls_replication_test.py

```diff
@@ -0,0 +1,102 @@
+    # In production, CA should be generated by a third party authority
+    # Expires in one day and is not encrtypted (-nodes)
+    # X.509 format for the key
+    ca_key = df_instance.dfly_path + "ca-key.pem"
+    ca_cert = df_instance.dfly_path + "ca-cert.pem"
+    step1 = rf'openssl req -x509 -newkey rsa:4096 -days 1 -nodes -keyout {ca_key} -out {ca_cert} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=AcmeStudios/CN=Gr/emailAddress=acme@gmail.com"'
+    subprocess.run(step1, shell=True)
+
+    # Step 2
+    # Generate Dragonfly's private key and certificate signing request (CSR)
+    tls_server_key = df_instance.dfly_path + tls_server_key_file_name
+    tls_server_req = df_instance.dfly_path + "df-req.pem"
+    step2 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_server_key} -out {tls_server_req} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"'
+    subprocess.run(step2, shell=True)
+
+    # Step 3
+    # Use CA's private key to sign dragonfly's CSR and get back the signed certificate
+    tls_server_cert = df_instance.dfly_path + tls_server_cert_file_name;
+    step3 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
+    subprocess.run(step3, shell=True)
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("t_master, t_replica", replication_cases)
+@dfly_args({"dbfilename": "test-tls-replication-{{timestamp}}"})
+async def test_replication_all(df_local_factory, df_seeder_factory, t_master, t_replica):
+    # 1. Spin up dragonfly without tls, debug populate and then shut it down
+    master = df_local_factory.create(port=BASE_PORT, proactor_threads=t_master)
+    master.start()
+    c_master = aioredis.Redis(port=master.port)
+    await c_master.execute_command("DEBUG POPULATE 100")
+    await c_master.execute_command("SAVE")
+    db_size = await c_master.execute_command("DBSIZE")
+    assert 100 == db_size
+
+    master.stop()
+    with_tls_args = tls_args(master)
+    gen_tls_cert(master)
+    # 2. Spin up dragonfly again, this time, with different options:
+    #   a. Enable tls
+    #   b. Allow non-tls connection from replica to master
+    master = df_local_factory.create(admin_port=ADMIN_PORT, admin_nopass=True, **with_tls_args, port=BASE_PORT, proactor_threads=t_master)
+    master.start()
+    # we need to sleep, because currently tls connections on master fail
+    # TODO Fix this once master properly loads tls certificates and redis-cli does not
+    # require --insecure flag, therefore for now we can't use `wait_available`
+    time.sleep(10)
+
+    # 3. Try to connect with redis cli on master. This should fail since redis-cli does
+    # not use tls key and certificate
+    c_master = aioredis.Redis(port=master.port)
+    try:
+        await c_master.execute_command("DBSIZE")
+        raise "Non tls connection connected on master with tls. This should NOT happen"
+    except redis.ConnectionError:
+        pass
+
```

Comment:
lets add a connection via admin port and run DBSIZE command, see that it passes

Reply:
this can also be a test in connection_test.py as checking that running command via admin port does not require tls is not related to replication

Reply:
I ended up having it here. It's not related to replication but it doesn't require to create a new test case for that. It's small detail ;)

Reply:
hmm, I kind of agree with Adi here. The feature here is being able to connect to the admin port without using the tls. I think the slave part is unnecessary machinery for this test.

Reply:
Maybe I missed something but @adiholden asked to promt admin port directly (on previous commits that was not the case because of how I initially implemented this) -- which I addressed.

Now for (`# 2. Try to connect on master without admin port. This should fail.`), I think it's good to be here because it asserts that the changes/flags introduced that allowed non-tls connections on admin port did not also allow non-tls connections on other ports (it could be the case that I messed up the if conditions here: https://github.com/dragonflydb/dragonfly/pull/1490/files#diff-b88cbca1b2c9337ab2a37262f9d516b117f4996da25e2ae173e0e129f2b82b20R310) -- this guards as against that potential mistake and that's why I put it here.

Reply:
The comment here is that the changes in dfly code are not specific for replication, we enable connecting from admin port without tls. Therefor we would like to test this specifically and the replication test is another test to check that replication work with this, we can add this test but f.e if replication with no tls breaks this does not mean that connecting via admin port with no tls fails.
f.e if the PR was about adding admin port to dfly I would test it in connection test, it is importatnt to validate that replication as well works with admin port there fore I will add another test for replication with admin port
I believe that we need